### DISCUSSION TITLE
Remote video via streaming protocols such as HLS or DASH

### DISCUSSION THREAD

**Discussion by thienphuoc** [axmolengine/axmol#1937] [Q&A] [answered]
HI all,

Do we currently support playing remote video via streaming protocols such as HLS or DASH?

I am testing the code:

`_videoPlayer->setURL("https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8");`

But it doesn't seem to be working.

**Comment by rh101** [comment]
Have you checked the remote video format? The supported formats are listed here: https://github.com/axmolengine/axmol/wiki/Media-Player

**Reply by thienphuoc** [reply]
I've tested with different video URLs on iOS, but they seem not to work.

I also took some MP4 videos, not HLS, from the page:

https://test-videos.co.uk/bigbuckbunny/mp4-h265

The videos I used are:

https://test-videos.co.uk/vids/bigbuckbunny/mp4/h265/360/Big_Buck_Bunny_360_10s_28MB.mp4

https://test-videos.co.uk/vids/bigbuckbunny/mp4/h265/1080/Big_Buck_Bunny_1080_10s_1MB.mp4

I also tried embedding some URLs like: https://www.youtube.com/

But most of them don't work.

I've come up with a temporary solution:

```
std::string htmlString = R"(
    <!DOCTYPE html>
    <html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        <title>HLS Video Player</title>
        <link href="https://vjs.zencdn.net/7.15.4/video-js.css" rel="stylesheet" />
    </head>
    <body>

    <video-js id="my-video" class="vjs-default-skin" controls preload="auto" width="640" height="360">
        <source src="https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8" type="application/x-mpegURL">
    </video-js>

    <script src="https://vjs.zencdn.net/7.15.4/video.js"></script>
    <script>
        var player = videojs('my-video');
        player.play();
    </script>

    </body>
    </html>
    )";
    _webView->loadHTMLString(htmlString);
    
```
This method is currently using a webview. Could it be that our native video player has some bugs and needs improvements?

**Reply by GamesStudioTwo** [reply]
Hi there this may not help, did you test on hardware or on Emulator, because if you run some audio formats on emu they done work" I have not done it with video format" so this may not help but emu give no sound on some format. but on hardware it works perfect with every format, I spent 2 weeks going nuts over this then I accidently compiled to hardware one day and found this out the hard way.

**Reply by thienphuoc** [reply, marked as answer]
It's truly amazing; today I ran it again on version 2.1.3 and it seems to be resolved.

```
   ax::ui::VideoPlayer*  _videoPlayer = ax::ui::VideoPlayer::create();
    _videoPlayer->setPosition(winSize / 2);
    _videoPlayer->setAnchorPoint(Vec2::ANCHOR_MIDDLE);
    _videoPlayer->setContentSize({800,600});
    _videoPlayer->setURL("https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8");
    _videoPlayer->play();
    this->addChild(_videoPlayer);
```

It is working well. I think you fixed it, right @rh101 ?

![Screenshot 2024-06-03 at 10 35 50](https://github.com/axmolengine/axmol/assets/17909394/ebde0d59-8dbb-4426-ac61-26fb27aa5051)


Thank you very much.

**Reply by rh101** [reply]
> It's truly amazing; today I ran it again on version 2.1.3 and it seems to be resolved.
> 
> ```
>    ax::ui::VideoPlayer*  _videoPlayer = ax::ui::VideoPlayer::create();
>     _videoPlayer->setPosition(winSize / 2);
>     _videoPlayer->setAnchorPoint(Vec2::ANCHOR_MIDDLE);
>     _videoPlayer->setContentSize({800,600});
>     _videoPlayer->setURL("https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8");
>     _videoPlayer->play();
>     this->addChild(_videoPlayer);
> ```
> 
> It is working well. I think you fixed it, right @rh101 ?
> 
> Thank you very much.

No, sorry, I can't take any credit for that.  The changes I made didn't have anything to do with the video media codec support.  Regardless of that, I'm glad it's now working for you.

**Reply by halx99** [reply]
Note: For android, we upgrade androidx.media3 from 1.0.2 to 1.2.1 in axmol-2.1.3
